helix-core/src/main/java/org/apache/helix/controller/rebalancer/DelayedAutoRebalancer.java

```diff
@@ -194,6 +200,17 @@ public IdealState computeNewIdealState(String resourceName,
     return idealState;
   }
 
+  private static List<String> filterOutOnOperationInstances(Map<String, InstanceConfig> instanceConfigMap,
+      Set<String> nodes) {
+    return  nodes.stream()
+        .filter(instance ->
+            !(instanceConfigMap.get(instance).getInstanceOperation().equals(
+            InstanceConstants.InstanceOperation.EVACUATE.name())||
+            instanceConfigMap.get(instance).getInstanceOperation().equals(
+            InstanceConstants.InstanceOperation.SWAP_IN.name())))
+        .collect(Collectors.toList());
```

Comment:
For simplicity, can we have a Set<InstanceOperation> and just check the Set.contains(instanceOperation)?

It is not scalable if you add more operations and the code looks huge.